research/picai/fl_nnunet/start_client.py

```diff
@@ -69,7 +69,7 @@ def main(
         # BaseClient Args
         device=DEVICE,
         metrics=metrics,
-        progress_bar=True,
+        progress_bar=False,
```

Comment:
I am not sure if this has anything to do with the issues that are popping up but I noticed the the number of processes to be used by the data augmenter. It is being set in this function in [nnUNetv2](https://github.com/MIC-DKFZ/nnUNet/blob/master/nnunetv2/utilities/default_n_proc_DA.py#L5). I am using a small machine to test and the call to os.cpu_count() reports much higher than what I have and even the default of 12 is more than the logical cpus I have. With the additional overhead of the server process, and eventually more than 1 client\, we need to make sure this is being set appropriately. I suggest you add a command line argument that allows you to pass the number of processes explicitly  and sets the environment variable (`$nnUNet_n_proc_DA`) that is leveraged in the linked function.

Reply:
I can confirm that the number of processes spawned by the augmenters is not causing the OOM issues as that was one of the first things I checked. PyTorch itself launches way more processes under the hood for reasons I don't understand, but I think it has something to do with the model being compiled. That being said, I agree that the number of processes used for augmentation should be a configurable parameter. Not sure how os.cpu_count() would report more than you have though? Is it possible that you have some high performance cpu's that are being counted as multiple cpu's?

Reply:
I had 2 cpus with 4 cores each with hyper-threading so 16 logical cpus and os.cpu_count() was returning 32. I am not sure if its returning all of the cpus on a given node or what. 

Thanks for adding though, this will be really helpful when we are doing FL with more than 1 client anyways